deepomatic/api/http_helper.py

```diff
@@ -209,15 +209,29 @@ def recursive_json_dump(prefix, obj, data_dict, omit_dot=False):
         return new_dict
 
     def send_request(self, requests_callable, *args, **kwargs):
-        # requests_callable must be a method from the requests module
-
         # this is the timeout of requests module
         requests_timeout = kwargs.pop('timeout', self.requests_timeout)
+
+        files = kwargs.pop('files', None)
+        if files:
+            for f in files.values():
+                # seek files before each retry
+                if hasattr(f, 'seek'):
+                    f.seek(0)
```

Comment:
what about non-seekable files? maybe we should fail instead of keeping the current bug of sending empty data?

Related issue in google storage python client: https://github.com/googleapis/google-resumable-media-python/issues/165
they mandate seekable streams. (that being said I remember seeing in their code that they memory buffered everything that is in flight and not yet confirmed by API, via application-level multi part uploads, so it's strange... maybe it has changed?)

or maybe it doesn't make sense for the `files` argument to have non seekable inputs; but the scenario should apply  to other ways of passing input data: `data`, `stream` arguments.

Reply:
Which non seekable files are you refering to ? Overall either those are bytes or stream that once read need to be reseeked.
I don't think the bug will still be there bug after this PR.

I don't want to break the current behavior now, though we can think about it (I would like also to remove the magic https://github.com/Deepomatic/deepomatic-client-python/issues/33)

Reply:
for reference, after meeting with hugo and some tests:
- `files` accepts non-streamable too, the real constraint is: if it's streamable and non seekable, then we cannot support retries: raise error on second try (maybe on first too?)
  - also need to keep supporting non streamable objects: strings, bytes, bytearrays, etc.
- `data` accepts generator/iterator, or a dict of such thing: hard to check all that for now. maybe types constraints will help.